src/main/kotlin/com.bugsnag.android.gradle/BugsnagInstallJniLibsTask.kt

```diff
@@ -60,5 +74,93 @@ open class BugsnagInstallJniLibsTask @Inject constructor(
+                        project.tasks.register<BugsnagInstallJniLibsTaskGradle66Plus>(name, configurationAction)
+                    }
+                    else -> {
+                        project.tasks.register<BugsnagInstallJniLibsTaskGradle6Plus>(name, configurationAction)
+                    }
+                  }
+              }
+              gradleVersion >= GradleVersions.VERSION_5_3 -> {
+                  project.tasks.register<BugsnagInstallJniLibsTask53Plus>(name, configurationAction)
+              }
+              else -> {
+                  project.tasks.register<BugsnagInstallJniLibsTaskLegacy>(name, configurationAction)
+              }
+            }
+        }
     }
 }
+
+/**
+ * Legacy [BugsnagInstallJniLibsTask] task that requires using [getProject] and
+ * [ProjectLayout.configurableFiles].
+ */
+internal open class BugsnagInstallJniLibsTaskLegacy @Inject constructor(
+    objects: ObjectFactory,
+    projectLayout: ProjectLayout
+) : BugsnagInstallJniLibsTask(objects, projectLayout) {
+    @Suppress("DEPRECATION") // Here for backward compatibility
+    @get:InputFiles
+    override val bugsnagArtifacts: ConfigurableFileCollection = projectLayout.configurableFiles()
+
+    override fun copy(action: (CopySpec) -> Unit): WorkResult = project.copy(action)
+    override fun zipTree(file: File): FileTree = project.zipTree(file)
+}
+
+/** Legacy [BugsnagInstallJniLibsTask] task that requires using [getProject]. */
+internal open class BugsnagInstallJniLibsTask53Plus @Inject constructor(
+    objects: ObjectFactory,
+    projectLayout: ProjectLayout
+) : BugsnagInstallJniLibsTask(objects, projectLayout) {
+    @get:InputFiles
+    override val bugsnagArtifacts: ConfigurableFileCollection = objects.fileCollection()
+
+    override fun copy(action: (CopySpec) -> Unit): WorkResult = project.copy(action)
+    override fun zipTree(file: File): FileTree = project.zipTree(file)
+}
+
+/** A Gradle 6+ compatible [BugsnagInstallJniLibsTask], which uses [FileSystemOperations]. */
+internal open class BugsnagInstallJniLibsTaskGradle6Plus @Inject constructor(
+    objects: ObjectFactory,
+    projectLayout: ProjectLayout,
+    private val fsOperations: FileSystemOperations
+) : BugsnagInstallJniLibsTask(objects, projectLayout) {
+    @get:InputFiles
+    override val bugsnagArtifacts: ConfigurableFileCollection = objects.fileCollection()
+
+    override fun copy(action: (CopySpec) -> Unit): WorkResult = fsOperations.copy(action)
+    override fun zipTree(file: File): FileTree = project.zipTree(file)
+}
+
+/**
+ * A Gradle 6.6+ compatible [BugsnagInstallJniLibsTask], which uses [FileSystemOperations]
+ * and [ArchiveOperations] to support configuration caching.
+ */
+internal open class BugsnagInstallJniLibsTaskGradle66Plus @Inject constructor(
+    objects: ObjectFactory,
+    projectLayout: ProjectLayout,
+    private val fsOperations: FileSystemOperations,
+    private val archiveOperations: ArchiveOperations
```

Comment:
This is what necessitated Gradle 6.6. May be worth adding a 6.6 matrix entry in tests

Reply:
I've created an internal ticket to track this - running against 6.6 and AGP 4.2.0 canary seems like a sensible idea.